java-squid/src/main/java/org/sonar/java/ast/parser/TreeFactory.java

```diff
@@ -230,38 +216,25 @@ public TypeTree newType(TypeTree basicOrClassType,
     }
   }
 
-  public TypeArgumentListTreeImpl newTypeArgumentList(InternalSyntaxToken openBracketToken, Tree typeArgument, Optional<List<AstNode>> rests, InternalSyntaxToken closeBracketToken) {
+  public TypeArgumentListTreeImpl newTypeArgumentList(InternalSyntaxToken openBracketToken, Tree typeArgument, Optional<List<Tuple<InternalSyntaxToken, Tree>>> rests, InternalSyntaxToken closeBracketToken) {
     ImmutableList.Builder<Tree> typeArguments = ImmutableList.builder();
-    List<AstNode> children = Lists.newArrayList();
-
     typeArguments.add(typeArgument);
-    children.add((AstNode) typeArgument);
 
     if (rests.isPresent()) {
-      for (AstNode rest : rests.get()) {
-        for (AstNode child : rest.getChildren()) {
-          // FIXME SONARJAVA-547 comma should be part of the ArgumentList as token
-          if (!child.is(JavaPunctuator.COMMA)) {
-            typeArguments.add((Tree) child);
-          }
-
-          children.add(child);
-        }
+      for (Tuple<InternalSyntaxToken, Tree> rest : rests.get()) {
+        // FIXME SONARJAVA-547 comma (rest.first()) should be part of the ArgumentList as token
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Take the required action to fix the issue indicated by this comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1134)